index.d.ts

```diff
@@ -1,57 +1,1324 @@
+    disabled?: boolean;
+    error?: string;
+    floating?: boolean;
+    hint?: string;
+    icon?: React.ReactNode | string;
+    label?: string;
+    maxLength?: number;
+    multiLine?: boolean;
+    name?: string;
+    onBlur?: React.FocusEventHandler;
+    onChange?: React.FormEventHandler;
+    onFocus?: React.FocusEventHandler;
+    onKeyPress?: React.MouseEventHandler;
+    required?: boolean;
+    theme?: InputThemeProps;
+    type?: string;
+    value?: any;
+  }
+  export class Input extends React.Component<InputProps, {}> {
+  }
+
+  // Layout
+  namespace Layout {
+    /**
+     * <Layout/>
+     */
+    interface LayoutThemeProps {
+      layout?: string;
+    }
+    interface LayoutProps extends Props {
+      children?: [NavDrawer | Panel | Sidebar];
+      theme?: LayoutThemeProps;
+    }
+    export class Layout extends React.Component<LayoutProps, {}> {
+    }
+
+    /**
+     * <NavDrawer/>
+     */
+    interface NavDrawerThemeProps {
+      active?: string;
+      drawerContent?: string;
+      lgPermangent?: string;
+      mdPermangent?: string;
+      navDrawer?: string;
+      pinned?: string;
+      scrim?: string;
+      scrollY?: string;
+      smPermanent?: string;
+      wide?: string;
+      xlPermanent?: string;
+      xxlPermangent?: string;
+      xxxlPermangent?: string;
+    }
+    interface NavDrawerProps extends Props {
+      active?: boolean;
+      children?: React.ReactNode;
+      onOverlayClick?: React.MouseEventHandler;
+      permanentAt?: "sm" | "md" | "lg" | "xl" | "xxl" | "xxxl";
+      pinned?: boolean;
+      scrollY?: boolean;
+      theme?: NavDrawerThemeProps;
+      width?: "normal" | "wide";
+    }
+    export class NavDrawer extends React.Component<NavDrawerProps, {}> {
+    }
+
+    /**
+     * <Panel/>
+     */
+    interface PanelThemeProps {
+      panel?: string;
+      scrollY?: string;
+    }
+    interface PanelProps extends Props {
+      children?: React.ReactNode;
+      scrollY?: boolean;
+      theme?: PanelThemeProps;
+    }
+    export class Panel extends React.Component<PanelProps, {}> {
+    }
+
+    /**
+     * <Sidebar/>
+     */
+    interface SidebarThemeProps {
+      pinned?: string;
+      scrollY?: string;
+      sidebar?: string;
+      sidebarContent?: string;
+    }
+    interface SidebarProps extends Props {
+      children?: React.ReactNode;
+      pinned?: boolean;
+      scrollY?: boolean;
+      theme?: SidebarThemeProps;
+      width?: number; // 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 25 | 33 | 50 | 66 | 75 | 100;
+    }
+    export class Sidebar extends React.Component<SidebarProps, {}> {
+    }
+
+  } // Layout
+
+  /**
+   * <Link/>
+   */
+  interface LinkThemeProps {
+    active?: string;
+    icon?: string;
+    link?: string;
+  }
+  interface LinkProps extends Props {
+    active?: boolean;
+    children?: React.ReactNode;
+    count?: number;
+    icon?: React.ReactNode | string;
+    label?: string;
+    theme?: LinkThemeProps;
+  }
```

Comment:
Not sure, but I think that the property 'href' which is implicit in the codebase, should be present here.